rpc/json/ws.go

```diff
@@ -33,13 +33,40 @@ func (wsc *wsConn) sendLoop() {
 	}
 }
 
+func CheckOrigin(r *http.Request, allowedOrigins []string) bool {
+	origin := r.Header.Get("Origin")
+	if origin == "" {
+		return true
```

Comment:
Are you sure? Why not `false`? How can we assure that this empty value is actually allowed/correct/secure? (see: https://stackoverflow.com/questions/42239643/when-do-browsers-send-the-origin-header-when-do-browsers-set-the-origin-to-null/42242802)